crates/pixi_build_frontend/tests/diagnostics.rs

```diff
@@ -144,12 +143,20 @@ async fn test_invalid_backend() {
     // connection.
     drop(in_tx);
 
-    let err = BuildFrontend::default()
-        .setup_protocol(SetupRequest {
-            source_dir: source_dir.path().to_path_buf(),
-            build_tool_override: ipc.into(),
-            build_id: 0,
-        })
+    let (workspace, package) = TomlManifest::from_toml_str(toml)
+        .unwrap()
+        .into_manifests(ExternalWorkspaceProperties::default())
+        .unwrap();
+    let protocol = pixi_build_frontend::pixi_protocol::ProtocolBuilder::new(
+        source_dir.path().to_path_buf(),
+        manifest.to_path_buf(),
+        workspace,
+        package.unwrap(),
+    )
+    .unwrap();
+
+    let err = protocol
+        .finish_with_tool(pixi_build_frontend::tool::Tool::Io(ipc), 0)
```

Comment:
I think we can simplify this even further by having a helper method directly in the pixi protocol implementation.

Reply:
Then you dont even have to go through the whole `Tool` type.

Reply:
Feel free to push to this PR! :)

Reply:
Done